core/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -191,6 +192,39 @@ public static AnalyzeResponse analyze(AnalyzeRequest request, String field, Anal
 
             analyzer = new CustomAnalyzer(tokenizerFactory.v1(), tokenizerFactory.v2(), charFilterFactories, tokenFilterFactories);
             closeAnalyzer = true;
+        } else if (request.normalizer() != null) {
+            // Get normalizer from indexAnalyzers
+            analyzer = indexAnalyzers.getNormalizer(request.normalizer());
+            if (analyzer == null) {
+                throw new IllegalArgumentException("failed to find normalizer under [" + request.normalizer() + "]");
+            }
+        } else if (((request.tokenFilters() != null && request.tokenFilters().size() > 0)
+                || (request.charFilters() != null && request.charFilters().size() > 0))) {
+            final IndexSettings indexSettings = indexAnalyzers == null ? null : indexAnalyzers.getIndexSettings();
+            // custom normalizer = if normalizer == null but filter or char_filter is not null and tokenizer/analyzer is null
+            // get charfilter and filter from request
+            CharFilterFactory[] charFilterFactories = new CharFilterFactory[0];
+            charFilterFactories = getCharFilterFactories(request, indexSettings, analysisRegistry, environment, charFilterFactories);
+            for (CharFilterFactory charFilter : charFilterFactories) {
+                if (charFilter instanceof MultiTermAwareComponent == false) {
+                    throw new IllegalArgumentException("Custom normalizer may not use char filter ["
+                        + charFilter.name() + "]");
+                }
+            }
+
+            TokenFilterFactory[] tokenFilterFactories = new TokenFilterFactory[0];
+            tokenFilterFactories = getTokenFilterFactories(request, indexSettings, analysisRegistry, environment, tokenFilterFactories);
+            for (TokenFilterFactory tokenFilter : tokenFilterFactories) {
+                if (tokenFilter instanceof MultiTermAwareComponent == false) {
+                    throw new IllegalArgumentException("Custom normalizer may not use filter ["
+                        + tokenFilter.name() + "]");
+                }
+            }
```

Comment:
looks like you are missing the call to `MultiTermAwareComponent.getMultiTermComponent`?